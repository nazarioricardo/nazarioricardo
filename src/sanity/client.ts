import { createClient, type ClientConfig } from "@sanity/client";

const date = new Date();

const config: ClientConfig = {
  projectId: process.env.NEXT_PUBLIC_SANITY_PROJECT_ID as string,
  dataset: "production",
  apiVersion: `${date.getFullYear()}-${date.getMonth() + 1}-${date.getDate()}`, // Usually you’d want to put in today’s date YYYY-MM-DD to get the newest version of the query API.
  useCdn: false,
};

const client = createClient(config);

export default client;
