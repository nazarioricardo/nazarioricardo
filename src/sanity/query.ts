import { groq } from "next-sanity";
import client from "./client";

export async function getProfile() {
  return client.fetch(
    groq`*[_type == "profile"]{
      _id,
      fullName,
      headline,
      profileImage {
        alt, 
        "image": asset->url
      },
      shortBio,
      location,
      fullBio,
      email,
      "resumeURL": resumeURL.asset->url,
      socialLinks,
      skills
    }`
  );
}

export async function getJob() {
  return client.fetch(
    groq`*[_type == "job"]{
      _id,
      name,
      jobTitle,
      "logo": logo.asset->url,
      url,
      description,
      startDate,
      endDate,
    }`
  );
}

export async function getProjects() {
  return client.fetch(
    groq`*[_type == "post"]{
      _id, 
      title,
      subtitle,
      description,
      "slug": slug.current,
      "logo": logo.asset->url,
    }`
  );
}

export async function getSingleProject(slug: string) {
  return await client.fetch(
    groq`*[_type == "post" && slug.current == $slug][0]{
      _id,
      title,
      body,
      subtitle,
      description,
    }`,
    { slug }
  );
}
