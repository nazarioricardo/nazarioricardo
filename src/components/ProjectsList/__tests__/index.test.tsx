import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { PortableTextObject } from "sanity";
import ProjectList from "../index";

const PLAIN_TEXT = "Plain text.";
const KEY: string = "R5FvMrjo";

const input: PortableTextObject = {
  _key: KEY,
  _type: "block",
  children: [
    {
      _key: "cZUQGmh4",
      _type: "p",
      marks: [],
      text: PLAIN_TEXT,
    },
  ],
  markDefs: [],
  style: "normal",
};

describe("ProjectList", () => {
  const mockProjects = [
    {
      _id: "1",
      image: "/path/to/image.jpg",
      title: "Project 1",
      description: "Description 1",
      slug: "project-1",
      body: [input],
      categories: ["react", "nextjs"],
    },
    {
      _id: "2",
      image: "/path/to/image.jpg",
      title: "Project 2",
      description: "Description 2",
      slug: "project-2",
      body: [input],
      categories: ["iOS", "AI"],
    },
  ];

  test("renders ProjectList component", () => {
    render(<ProjectList projects={mockProjects} />);
    mockProjects.forEach((project) => {
      expect(screen.getByText(project.title)).toBeInTheDocument();
      expect(screen.getByText(project.description)).toBeInTheDocument();
      const imgElement: HTMLImageElement = screen.getByAltText(
        "Image for " + project.title
      );
      expect(imgElement).toBeInTheDocument();
      expect(imgElement.src).toMatch(
        /\/_next\/image\?url=%2Fpath%2Fto%2Fimage\.jpg/
      );
    });
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<ProjectList projects={mockProjects} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
