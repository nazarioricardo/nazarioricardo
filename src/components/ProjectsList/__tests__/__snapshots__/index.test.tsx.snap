// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectList matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="projectList"
  >
    <a
      class="project"
      href="/projects/project-1"
    >
      <div
        class="imageContainer"
      >
        <img
          alt="Image for Project 1"
          class="image false"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=%2Fpath%2Fto%2Fimage.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fpath%2Fto%2Fimage.jpg&w=640&q=75 640w, /_next/image?url=%2Fpath%2Fto%2Fimage.jpg&w=750&q=75 750w, /_next/image?url=%2Fpath%2Fto%2Fimage.jpg&w=828&q=75 828w, /_next/image?url=%2Fpath%2Fto%2Fimage.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fpath%2Fto%2Fimage.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fpath%2Fto%2Fimage.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fpath%2Fto%2Fimage.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fpath%2Fto%2Fimage.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      <div
        class="content"
      >
        <h3>
          Project 1
        </h3>
        <p>
          Description 1
        </p>
      </div>
    </a>
    <a
      class="project"
      href="/projects/project-2"
    >
      <div
        class="imageContainer"
      >
        <img
          alt="Image for Project 2"
          class="image false"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=%2Fpath%2Fto%2Fimage.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fpath%2Fto%2Fimage.jpg&w=640&q=75 640w, /_next/image?url=%2Fpath%2Fto%2Fimage.jpg&w=750&q=75 750w, /_next/image?url=%2Fpath%2Fto%2Fimage.jpg&w=828&q=75 828w, /_next/image?url=%2Fpath%2Fto%2Fimage.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fpath%2Fto%2Fimage.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fpath%2Fto%2Fimage.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fpath%2Fto%2Fimage.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fpath%2Fto%2Fimage.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      <div
        class="content"
      >
        <h3>
          Project 2
        </h3>
        <p>
          Description 2
        </p>
      </div>
    </a>
  </section>
</DocumentFragment>
`;
